storage-node/src/services/sync/tasks.ts

```diff
@@ -77,39 +72,53 @@ export class DeleteLocalFileTask implements SyncTask {
  * Download the file from the remote storage node to the local storage.
  */
 export class DownloadFileTask implements SyncTask {
-  dataObjectId: string
-  expectedHash?: string
-  uploadsDirectory: string
-  tempDirectory: string
-  url: string
-  downloadTimeout: number
-  hostId: string
+  operatorUrls: string[]
 
   constructor(
-    baseUrl: string,
-    dataObjectId: string,
-    expectedHash: string | undefined,
-    uploadsDirectory: string,
-    tempDirectory: string,
-    downloadTimeout: number,
-    hostId: string
+    baseUrls: string[],
+    private dataObjectId: string,
+    private expectedHash: string,
+    private uploadsDirectory: string,
+    private tempDirectory: string,
+    private downloadTimeout: number,
+    private hostId: string
   ) {
-    this.dataObjectId = dataObjectId
-    this.expectedHash = expectedHash
-    this.uploadsDirectory = uploadsDirectory
-    this.tempDirectory = tempDirectory
-    this.downloadTimeout = downloadTimeout
-    this.url = urljoin(baseUrl, 'api/v1/files', dataObjectId)
-    this.hostId = hostId
+    this.operatorUrls = baseUrls.map((baseUrl) => urljoin(baseUrl, 'api/v1/files', dataObjectId))
   }
 
   description(): string {
-    return `Sync - downloading file: ${this.url} to ${this.uploadsDirectory} ....`
+    return `Sync - Trying for download of: ${this.dataObjectId} ....`
   }
 
   async execute(): Promise<void> {
+    const operatorUrlIndices: number[] = _.shuffle(_.range(this.operatorUrls.length))
+
+    for (const randomUrlIndex of operatorUrlIndices) {
+      const chosenBaseUrl = this.operatorUrls[randomUrlIndex]
+      logger.debug(`Sync - random storage node URL was chosen ${chosenBaseUrl}`)
+
+      // Remove random url from the original list.
+      _.remove(operatorUrlIndices, (index) => index === randomUrlIndex)
```

Comment:
removing the element is no longer necessary, even potentially problematic.

Reply:
Ah right. Missed that.

Reply:
Addressed in [3c02240](https://github.com/Joystream/joystream/pull/5014/commits/3c02240ef9fcf9a64945e94c880f33a16849de16)